deps/config/types.go

```diff
@@ -614,3 +627,31 @@ type Libp2pConfig struct {
 	// Format: multiaddress
 	NoAnnounceAddresses []string
 }
+
+type IPNIConfig struct {
+	// Enable set whether to enable indexing announcement to the network and expose endpoints that
+	// allow indexer nodes to process announcements. Enabled by default.
+	Enable bool
```

Comment:
`Disable` would make sense. This will completely disable everything. HTTP server is not part of this PR where this module will be called. When it is disabled, it will simply not create a provider. Thus no publishing. I should disable indexing jobs as well in case of disable. Don't make sense to create ads if we are not publishing.